src/exporter/exporter.go

```diff
@@ -130,20 +78,21 @@ func exposedMetricsMiddleware(metricsMiddleware []MetricMiddleware, promHandler
 
 // MustExportMetrics will read the config from mainConfigFile if any or use a default one.
 func MustExportMetrics(handlerEndpoint string, listenPort uint16, conf config.RootConfig) (srv *http.Server) {
-	if collector, err := newSkycoinCollector(conf); err != nil {
+	c := cache.NewCache()
+	if collector, err := newSkycoinCollector(c, conf); err != nil {
```

Comment:
Not `SkycoinCollector` ... name must be generic since it can be used to scrape anything , not just Skycoin